src/RichText/RichText.tsx

```diff
@@ -51,28 +66,62 @@ const defaultConfig = {
 	disableNativeSpellChecker: false
 };
 
-class RichText extends React.PureComponent {
+export interface RichTextProps {
+	/**
+	 * Root div's id
+	 */
+	id?: string;
+	/**
+	 * Current value
+	 */
+	value?: string;
+	/**
+	 * CKEditor configuration object
+	 */
+	config?: any;
+	/**
+	 * Event handlers
+	 */
+	events?: { [eventName: string]: (e) => any };
+	/**
+	 * URL of ckeditor.js file. It is loaded from CDN by default.
+	 */
+	scriptUrl?: string;
+	/**
+	 * Root div's class
+	 */
+	className?: string;
+	/**
+	 * Throttle delay in milliseconds
+	 */
+	throttle?: number;
+}
+
+export default class RichText extends React.PureComponent<RichTextProps> {
+	private mounted = false;
 	static defaultProps = {
 		id: "richText",
 		config: {},
+		events: {},
 		scriptUrl: "https://cdn.ckeditor.com/4.7.3/standard/ckeditor.js",
-		throttle: 500
+		throttle: 500,
+		withPlugins: true
 	};
 
 	static editorInstances = {};
 
 	componentDidUpdate() {
 		if (
-			RichText.editorInstances[this.props.id] &&
-			RichText.editorInstances[this.props.id].getData().trim() !== this.props.value
+			RichText.editorInstances[this.props.id!] &&
+			RichText.editorInstances[this.props.id!].getData().trim() !== this.props.value
 		) {
-			RichText.editorInstances[this.props.id].setData(this.props.value);
+			RichText.editorInstances[this.props.id!].setData(this.props.value);
 		}
 	}
 
 	componentDidMount() {
 		this.mounted = true;
-		if (!window.CKEDITOR) {
+		if (!window["CKEDITOR"]) {
```

Comment:
Maybe add typings to support old version?

Reply:
Added